pkg/controller/provider/controller.go

```diff
@@ -354,6 +387,22 @@ func (r *Reconciler) getSecret(provider *api.Provider) (*v1.Secret, error) {
 	return secret, nil
 }
 
+func (r *Reconciler) handleServerCreationFailure(provider *api.Provider, errMsg error) {
+	provider.Status.Phase = ServerCreationFailed
```

Comment:
why not to use `ConnectionFailed`?

Reply:
because I think its a different scenario that `ConnectionFailed`, we are failing to create the server, the connection failed should indicate about issue between the connectivity to the created pod. we don't have this extra steps for any one of the other providers but IMO it make sense to add new phases.

Reply:
I agree that the OVA provider has additional steps compared to the other providers but the question is whether we have to reflect them to users? it's a matter of perspective - let's say that the server pod doesn't start, ok? as a user, I get the same result - fail to connect to the collection of VMs to import

the addition of the extra steps doesn't come without a cost - we add phases, we break the general flow with OVA-specific things (some we have to have, like the deployment of the server pod's deployment). if it's not very useful for users, we should minimize those provider-specific things

 for debugging we have other ways to expose data - with logs or by setting messages in conditions in a way that better reflect the underlying error